babel.config.js

```diff
@@ -25,99 +25,394 @@
+ */
+const defaultFilterImportsPlugin = [
+  'filter-imports',
+  {
+    imports: {
+      './polyfills/document-contains': ['installDocContains'],
+      './polyfills/domtokenlist': ['installDOMTokenList'],
+      './polyfills/fetch': ['installFetch'],
+      './polyfills/math-sign': ['installMathSign'],
+      './polyfills/object-assign': ['installObjectAssign'],
+      './polyfills/object-values': ['installObjectValues'],
+      './polyfills/promise': ['installPromise'],
+      './polyfills/array-includes': ['installArrayIncludes'],
+      '../third_party/css-escape/css-escape': ['cssEscape'],
+      '../build/ampshared.css': ['cssText', 'ampSharedCss'],
+      '../build/ampdoc.css': ['cssText', 'ampDocCss'],
+    },
+  },
+];
+
+/**
+ * Default istanbul plugin. Used by tests.
+ */
+const defaultInstanbulPlugin = [
+  'istanbul',
+  {
+    exclude: [
+      'ads/**/*.js',
+      'build-system/**/*.js',
+      'extensions/**/test/**/*.js',
+      'third_party/**/*.js',
+      'test/**/*.js',
+      'testing/**/*.js',
+    ],
+  },
+];
+
+/**
+ * Gets relative paths to all the devDependencies defined in package.json.
+ *
+ * @return {!Array<string>}
+ */
+function devDependencies() {
+  const file = fs.readFileSync('package.json', 'utf8');
+  const packageJson = JSON.parse(file);
+  const devDependencies = Object.keys(packageJson['devDependencies']);
+  return devDependencies.map((p) => `./node_modules/${p}`);
+}
+
+/**
+ * Ignore devDependencies except for 'chai-as-promised' which contains ES6 code.
+ * ES6 code is fine for most test environments, but not for integration tests
+ * running on SauceLabs since some older browsers need ES5.
+ */
+const ignoredGlobalModules = devDependencies().filter(
+  (dep) => dep.indexOf('chai-as-promised') === -1
+);
+
+/**
+ * Resolves the full path of a local plugin with the given name.
+ *
+ * @param {string} name
+ * @return {string}
+ */
+function localPlugin(name) {
+  return require.resolve(`./build-system/babel-plugins/babel-plugin-${name}`);
+}
+
+/**
+ * Resolves the full path of a babel monorepo plugin with the given name.
+ *
+ * @param {string} name
+ * @return {string}
+ */
+function babelPlugin(name) {
+  return `@babel/plugin-${name}`;
+}
 
-const targets = (esm) => {
-  if (esm) {
-    return {'esmodules': true};
+/**
+ * Computes options for the minify-replace plugin
+ *
+ * @return {Array<string|Object>}
+ */
+function getReplacePlugin() {
+  /**
+   * @param {string} identifierName the identifier name to replace
+   * @param {boolean} value the value to replace with
+   * @return {!Object} replacement options used by minify-replace plugin
+   */
+  function createReplacement(identifierName, value) {
+    return {
+      identifierName,
+      replacement: {type: 'booleanLiteral', value: !!value},
+    };
   }
 
-  if (isTravisBuild()) {
-    return {'browsers': ['Last 2 versions', 'safari >= 9']};
+  const replacements = [createReplacement('IS_ESM', argv.esm)];
```

Comment:
Am I correct in saying that only when `--esm` is applied via CLI we get ESM transforms? 

Does this preclude testing documents that have both the `module` and `nomodule` builds on them with a mix of devices and capabilities?